executor.go

```diff
@@ -113,13 +115,13 @@ func (e *executor) start() {
 		case <-e.stopCh:
 			cancel()
 			runningJobsWg.Wait()
-			e.stopCh <- struct{}{}
+			close(e.stoppedCh)
 			return
 		}
 	}
 }
 
 func (e *executor) stop() {
-	e.stopCh <- struct{}{}
-	<-e.stopCh
+	close(e.stopCh)
+	<-e.stoppedCh
```

Comment:
We need two channels for this use case. In the previous implementation, `struct{}{}` sent at L123 might be received at L124, which means `start()` loop never stops.